src/messages/variant.rs

```diff
@@ -126,6 +128,21 @@ impl Variant {
                 let proof_chain = proof_chain.ok_or(RoutingError::InvalidMessage)?;
                 payload.verify(proof_chain, trusted_keys)
             }
+            Self::NeighbourInfo { elders_info, .. } => {
+                if !elders_info.self_verify() {
+                    return Err(RoutingError::FailedSignature);
+                }
+                let proof_chain = proof_chain.ok_or(RoutingError::InvalidMessage)?;
+                if !proof_chain.has_key(&elders_info.proof.public_key) {
+                    return Err(RoutingError::UntrustedMessage);
+                }
+
+                match proof_chain.check_trust(trusted_keys) {
+                    TrustStatus::Trusted => Ok(VerifyStatus::Full),
+                    TrustStatus::Unknown => Ok(VerifyStatus::Unknown),
+                    TrustStatus::Invalid => Err(RoutingError::UntrustedMessage),
+                }
```

Comment:
Can we replace all this with just `elders_info.verify(proof_chain, trusted_keys)` ?

Reply:
the current `EldersInfo::verify` function only check against the proof_chain.
the part of `proof_chain.check_trust(trusted_keys)` is actually nothing related to `EldersInfo`

Also `EldersInfo::self_verify` is currently a separate function

I can make all this as a combined verify function within EldersInfo, any name suggestion for it?
how about `self_and_proof_chain_check` ?

Reply:
Sorry, that's won't work. We should do something like this I think:
```rust
elders_info.verify(proof_chain)...
proof_chain.check_trust(trusted_keys)...
```

Reply:
Then that will be almost the same as the current ?

Reply:
maybe just change
```rust
 if !proof_chain.has_key(&elders_info.proof.public_key) {
                    return Err(RoutingError::UntrustedMessage);
                }
```
to
```rust
 if !elders_info.verify(proof_chain) {
      return Err(RoutingError::UntrustedMessage);
 }
```

Reply:
Yeah, I mean the two calls: `elders_info.self_verify` and `proof_chain.has_key(elders_info.proof.public_key)` can be replaced by just `elders_info.verify(proof_chain)`. The rest should stay.